### PR TITLE
[T6813]bad_connector_woocommerce:sync basic settings-BAD-PAA

### PR DESCRIPTION
**Pull request by bizzappdevteam**
Task Description:

- Get default Currency.

- Get default Weight and Dimensions.

- Get payment gateways and syn with direct binding object.

### CODE REVIEW

File: bad_connector_woocommerce/README.rst [hunk 1]
```diff
@@ -157,6 +157,9 @@
   - Added 'Tax Include' in field at backend level which get the setting of 'Tax Include'.
   - Added Condition on search tax base on 'Included in Price'.
 
+* Import Payment Gateways:
```
Comment (bizzappdevteam) [inline bad_connector_woocommerce/README.rst]: Improve readme as suggested

File: bad_connector_woocommerce/models/woo_settings/importer.py [hunk 2]
```diff
@@ -67,4 +67,30 @@ def _after_import(self, binding, **kwargs):
...
+                        "'%s' currency not found, ensure that currency is active!!!"
+                        % binding.value
+                    )
+                )
+            binding.backend_id.write({"currency_id": currency.id})
+
+        if binding.external_id == "woocommerce_weight_unit":
+            weight_uom = self.env["uom.uom"].search(
+                [("name", "=", binding.value)], limit=1
+            )
```
Comment (bizzappdevteam) [inline bad_connector_woocommerce/models/woo_settings/importer.py:87]: what if weight_uom is not found while searching

File: bad_connector_woocommerce/models/woo_settings/importer.py [hunk 3]
```diff
@@ -67,4 +67,30 @@ def _after_import(self, binding, **kwargs):
...
+            binding.backend_id.write({"weight_uom_id": weight_uom.id})
+
+        if binding.external_id == "woocommerce_dimension_unit":
+            dimension_uom = self.env["uom.uom"].search(
+                [("name", "=", binding.value)], limit=1
+            )
```
Comment (bizzappdevteam) [inline bad_connector_woocommerce/models/woo_settings/importer.py:93]: what if dimension_uom  not found

File: bad_connector_woocommerce/views/woo_payment_gateway_views.xml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                        </group>
+                        <group>
+                            <field name="backend_id" widget="selection" />
+                            <field name="external_id" />
+                        </group>
+                    </group>
+                    <notebook>
+                        <page string="General Information" name="general_information">
+                            <group string="Internal Notes">
+                                <field
```
Comment (bizzappdevteam) [inline bad_connector_woocommerce/views/woo_payment_gateway_views.xml]: we don't need General Information page so simply add a description field

File: bad_connector_woocommerce/README.rst [hunk 5]
```diff
@@ -96,6 +96,16 @@
...
+    - Navigate to Woocommerce Backends by going to Connectors > WooCommerce > WooCommerce Backends.
+    - Add Backend Credentials to Import Metadata which contains Payment Gateways.
+    - Click the 'Sync Metadata' button to import Payment Gateways from WooCommerce.
+
+* Import Currency and UOM:
+    - Navigate to Woocommerce Backends by going to Connectors > WooCommerce > WooCommerce Backends.
+    - Add Backend Credentials to Import Metadata which contains default currency, Default Weight and Dimension.
+    - Click the 'Sync Metadata' button to import default currency, Default Weight and Dimension mesurement.
+
 **Usage**
```
Comment (bizzappdevteam) [inline bad_connector_woocommerce/README.rst:110]: update readme properly

File: bad_connector_woocommerce/tests/test_import_payment_gateway.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+            "Direct bank transfer",
+            "Payment name is not matched with response!",
+        )
+        self.assertEqual(
+            payment1.slug,
+            "Make your payment directly into our bank account.",
+            "Payment slug is not matched with response!",
+        )
+        self.assertEqual(
+            payment1.enable,
```
Comment (bizzappdevteam) [inline bad_connector_woocommerce/tests/test_import_payment_gateway.py]: use assertTrue for boolean

File: bad_connector_woocommerce/tests/fixtures/cassettes/import_woo_currency.yaml [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+                "group_id": "products",
+                "_links": {
+                    "self": [
+                        {
+                            "href": "http://localhost:8080/wp-json/wc/v3/settings/products/woocommerce_dimension_unit"
+                        }
+                    ],
+                    "collection": [
+                        {
+                            "href": "http://localhost:8080/wp-json/wc/v3/settings/products"
```
Comment (bizzappdevteam) [inline bad_connector_woocommerce/tests/fixtures/cassettes/import_woo_currency.yaml]: correct this URL wherever required.

File: bad_connector_woocommerce/components/backend_adapter.py [hunk 8]
```diff
@@ -186,6 +186,9 @@ class GenericAdapter(AbstractComponent):
     _apply_on = "woo.backend"
     _last_update_date = "date_modified"
     _woo_model = None
+    _woo_default_currency = None
+    _woo_default_weight = None
+    _woo_default_dimension = None
```
Comment (bizzappdev) [inline bad_connector_woocommerce/components/backend_adapter.py]: better to pass this by inheriting search method in kwargs and have condition based on it

File: bad_connector_woocommerce/models/woo_settings/importer.py [hunk 9]
```diff
@@ -67,4 +67,44 @@ def _after_import(self, binding, **kwargs):
             else:
                 include_tax = False
             binding.backend_id.write({"include_tax": include_tax})
+
+        if binding.external_id == "woocommerce_currency":
+            currency = self.env["res.currency"].search(
+                [("name", "=", binding.value)], limit=1
```
Comment (bizzappdev) [inline bad_connector_woocommerce/models/woo_settings/importer.py:73]: best will be based on code.

File: bad_connector_woocommerce/models/woo_settings/importer.py [hunk 10]
```diff
@@ -67,4 +67,44 @@ def _after_import(self, binding, **kwargs):
...
+                        % binding.value
+                    )
+                )
+            binding.backend_id.write({"currency_id": currency.id})
+
+        if binding.external_id == "woocommerce_weight_unit":
+            weight_uom = self.env["uom.uom"].search(
+                [("name", "=", binding.value)], limit=1
+            )
+            if not weight_uom:
```
Comment (bizzappdev) [inline bad_connector_woocommerce/models/woo_settings/importer.py]: Lets discuss this

File: bad_connector_woocommerce/models/woo_settings/importer.py [hunk 11]
```diff
@@ -67,4 +67,44 @@ def _after_import(self, binding, **kwargs):
...
+                        % binding.value
+                    )
+                )
+            binding.backend_id.write({"weight_uom_id": weight_uom.id})
+
+        if binding.external_id == "woocommerce_dimension_unit":
+            dimension_uom = self.env["uom.uom"].search(
+                [("name", "=", binding.value)], limit=1
+            )
+            if not dimension_uom:
```
Comment (bizzappdev) [inline bad_connector_woocommerce/models/woo_settings/importer.py]: lets discuss

**Review by bizzappdev — CHANGES_REQUESTED** [review]

File: bad_connector_woocommerce/models/woo_settings/importer.py (hunk 9 above)
Comment (bizzappdevteam) [reply bad_connector_woocommerce/models/woo_settings/importer.py:73]: The name field in res.currency actually consist the code of currency.

File: bad_connector_woocommerce/components/backend_adapter.py [hunk 12]
```diff
@@ -194,6 +194,31 @@ def search(self, filters=None, **kwargs):
         result = self._call(
             resource_path=self._woo_model, arguments=filters, http_method="get"
         )
+
+        if "_woo_default_currency" in kwargs:
```
Comment (bizzappdev) [inline bad_connector_woocommerce/components/backend_adapter.py]:
```suggestion
        if kwargs.get("_woo_default_currency", False):
```

**Comment by bizzappdevteam** [comment]
This PR is merged in #58.

**Pull request closed without merging**